src/FakeItEasy/Configuration/BuildableCallRule.cs

```diff
@@ -11,7 +11,7 @@ namespace FakeItEasy.Configuration
     internal abstract class BuildableCallRule
         : IFakeObjectCallRule
     {
-        private readonly List<WherePredicate> wherePredicates;
+        private readonly SynchronizedCollection<WherePredicate> wherePredicates;

```

Comment:
I won't be "that guy" and argue it with you; I'll revert and we can agree to disagree on this one. I think there is value in certain types of framework going a little further than the minimum required - unit testing and mocking being among those.